### PR TITLE
feat: remove PRISMA_FMT_BINARY references

### PR DESCRIPTION
**Pull request by jkomyno**
DO NOT MERGE UNTIL `prisma@4.10.0` is ready to be released.

## Describe this PR

Follow-up for https://github.com/prisma/prisma/pull/16923.
We've stopped using the `prisma-fmt` formatter binary in `prisma@4.3.0`. Although we're currently still downloading the binary, we're no longer using it. As a result, `PRISMA_FMT_BINARY` has been a useless environment variable for months.

## Changes

Remove references to `PRISMA_FMT_BINARY`.

## What issue does this fix?

Contributes to https://github.com/prisma/prisma/issues/16924

### CODE REVIEW

**Comment by jkomyno** [comment]
Note: @Jolg42 and I had a doubt about specifying the behavior of `PRISMA_FMT_BINARY` with an asterisk pointing out that it's no longer in use as of `prisma@4.3.0`. However, we figured that custom Prisma binaries is such an edge case scenario that it's probably not worth it to keep mentions of `PRISMA_FMT_BINARY` around.

**Review by ruheni — APPROVED** [review]

**Comment by janpio** [comment]
Where did the documentation of the env var move for users of older Prisma versions?
We only have one docs website, so we need to make sure it also works for other versions.
In this case we could for example add a "In older versions" headline that has a minimal mention of what the env var used to to and be with info when we removed it.

**Comment by janpio** [comment]
The other PR https://github.com/prisma/docs/pull/4495/files for INtrospection Engine also updates some `prisma -v` output and serverless framework instructions from the docs which seems to have been skipped here.

File: content/400-reference/200-api-reference/300-environment-variables-reference.mdx [hunk 1]
```diff
@@ -186,11 +186,7 @@ PRISMA_MIGRATION_ENGINE_BINARY=custom/my-migration-engine-unix
 PRISMA_INTROSPECTION_ENGINE_BINARY=custom/my-introspection-engine-unix
 ```
 
-<Admonition type="warning">
-
-- The Introspection Engine is served by the Migration Engine from [4.9.0](https://github.com/prisma/prisma/releases/tag/4.9.0). Therefore, the `PRISMA_INTROSPECTION_ENGINE` environment variable will not be used.
-
-</Admonition>
```
Comment (janpio) [inline content/400-reference/200-api-reference/300-environment-variables-reference.mdx:193]: Why his unrelated change here?

File: content/400-reference/200-api-reference/300-environment-variables-reference.mdx [hunk 2]
```diff
@@ -200,7 +196,11 @@ PRISMA_INTROSPECTION_ENGINE_BINARY=custom/my-introspection-engine-unix
 PRISMA_FMT_BINARY=custom/my-custom-format-engine-unix
 ```
 
-### CLI Binary Targets
```
Comment (janpio) [inline content/400-reference/200-api-reference/300-environment-variables-reference.mdx:203]: Why was this headline deleted?

File: content/400-reference/200-api-reference/300-environment-variables-reference.mdx [hunk 3]
```diff
@@ -186,11 +186,7 @@ PRISMA_MIGRATION_ENGINE_BINARY=custom/my-migration-engine-unix
 PRISMA_INTROSPECTION_ENGINE_BINARY=custom/my-introspection-engine-unix
 ```
 
-<Admonition type="warning">
-
-- The Introspection Engine is served by the Migration Engine from [4.9.0](https://github.com/prisma/prisma/releases/tag/4.9.0). Therefore, the `PRISMA_INTROSPECTION_ENGINE` environment variable will not be used.
-
-</Admonition>
+### CLI Binary Targets
```
Comment (janpio) [inline content/400-reference/200-api-reference/300-environment-variables-reference.mdx:189]: Why move this headline up here? `PRISMA_FMT_BINARY` has nothing to do with "CLI Binary Targets".

File: content/400-reference/200-api-reference/300-environment-variables-reference.mdx [hunk 4]
```diff
@@ -186,11 +186,7 @@ PRISMA_MIGRATION_ENGINE_BINARY=custom/my-migration-engine-unix
 PRISMA_INTROSPECTION_ENGINE_BINARY=custom/my-introspection-engine-unix
 ```
 
-<Admonition type="warning">
```
Comment (janpio) [inline content/400-reference/200-api-reference/300-environment-variables-reference.mdx:189]: Why remove this unrelated admonition about a different env var? The fmt change does not invalidate this.

File: content/400-reference/200-api-reference/300-environment-variables-reference.mdx (hunk 3 above)
Comment (ruheni) [reply content/400-reference/200-api-reference/300-environment-variables-reference.mdx:189]: I must have made a mistake when applying the feedback. I'll make the change in another PR.

Comment (ruheni) [reply content/400-reference/200-api-reference/300-environment-variables-reference.mdx:189]: Should be covered by https://github.com/prisma/docs/pull/4502

File: content/400-reference/200-api-reference/300-environment-variables-reference.mdx (hunk 4 above)
Comment (ruheni) [reply content/400-reference/200-api-reference/300-environment-variables-reference.mdx:189]: https://github.com/prisma/docs/pull/4502 fixes this
